### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by dashpole**
Update Otel versions:

* stable opentelemetry-go: v1.3.0 -> v1.6.2
* unstable opentelemetry-go= v.26.0 -> v0.28.0
* unstable opentelemetry-go-contrib= v0.31.0
* collector core= v0.46.0 -> v0.48.0
* collector contrib=v0.40.0 -> v0.48.0

This adapts our examples to the (hopefully stable) otel API.

This also updates the go version to 1.17, and updates almost all other dependencies.  This excludes the tools directory, which seems non-trivial to update, and does not update `github.com/googleinterns/cloud-operations-api-mock`, as the update breaks tests.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -2,7 +2,7 @@ version: 2
 jobs:
   build:
     docker:
-      - image: circleci/golang:1.16
+      - image: circleci/golang:1.17
```
Comment (aabmass) [inline .circleci/config.yml:5]: Is this the only version supported in Collector and Go SDK? Wondering if this is a considered a breakin change

File: exporter/collector/integrationtest/testcase.go [hunk 2]
```diff
@@ -95,8 +95,8 @@ func (m *MetricsTestCase) LoadOTLPMetricsInput(
 
 	for i := 0; i < metrics.ResourceMetrics().Len(); i++ {
 		rm := metrics.ResourceMetrics().At(i)
-		for i := 0; i < rm.InstrumentationLibraryMetrics().Len(); i++ {
-			rmi := rm.InstrumentationLibraryMetrics().At(i)
+		for i := 0; i < rm.ScopeMetrics().Len(); i++ {
+			rmi := rm.ScopeMetrics().At(i)
```
Comment (aabmass) [inline exporter/collector/integrationtest/testcase.go:99]: nit it looks like this has gone through a few renames, maybe `sms` for ScopeMetrics would be more appropriate?

File: exporter/collector/internal/datapointstorage/datapointcache.go [hunk 3]
```diff
@@ -81,7 +81,7 @@ func (c Cache) gc(shutdown <-chan struct{}, tickerCh <-chan time.Time) bool {
 }
 
 // Identifier returns the unique string identifier for a metric
-func Identifier(resource *monitoredrespb.MonitoredResource, extraLabels map[string]string, metric pdata.Metric, labels pdata.AttributeMap) string {
+func Identifier(resource *monitoredrespb.MonitoredResource, extraLabels map[string]string, metric pdata.Metric, labels pdata.Map) string {
```
Comment (aabmass) [inline exporter/collector/internal/datapointstorage/datapointcache.go:84]: same nit here, OTel used to call them labels and now they're "attributes". Might be worth updating, but up to you

File: exporter/collector/metrics.go [hunk 4]
```diff
@@ -179,12 +179,12 @@ func (me *MetricsExporter) PushMetrics(ctx context.Context, m pdata.Metrics) err
 	for i := 0; i < rms.Len(); i++ {
 		rm := rms.At(i)
 		monitoredResource, extraResourceLabels := me.mapper.resourceToMonitoredResource(rm.Resource())
-		ilms := rm.InstrumentationLibraryMetrics()
+		ilms := rm.ScopeMetrics()
```
Comment (aabmass) [inline exporter/collector/metrics.go:182]: same nit here, `sms` or similar may be more appropriate now and `s/ilm/sm` below?

File: exporter/collector/metrics.go [hunk 5]
```diff
@@ -319,7 +319,7 @@ func (me *MetricsExporter) createServiceTimeSeries(ctx context.Context, ts []*mo
 	)
 }
 
-func (m *metricMapper) instrumentationLibraryToLabels(il pdata.InstrumentationLibrary) labels {
+func (m *metricMapper) instrumentationScopeToLabels(il pdata.InstrumentationScope) labels {
```
Comment (aabmass) [inline exporter/collector/metrics.go:322]:
```suggestion
func (m *metricMapper) instrumentationScopeToLabels(is pdata.InstrumentationScope) labels {
```

**Review by aabmass — APPROVED** [review]
There's a bunch of places where local variable names are confusing because of pdata renames. It would probably be good to fix at some point but I don't wanna block this PR

File: exporter/collector/metrics_test.go [hunk 6]
```diff
@@ -1445,34 +1437,34 @@ func TestKnownDomains(t *testing.T) {
 	}
 }
 
-func TestInstrumentationLibraryToLabels(t *testing.T) {
-	newInstrumentationLibrary := func(name, version string) pdata.InstrumentationLibrary {
-		il := pdata.NewInstrumentationLibrary()
+func TestInstrumentationScopeToLabels(t *testing.T) {
+	newInstrumentationScope := func(name, version string) pdata.InstrumentationScope {
+		il := pdata.NewInstrumentationScope()
```
Comment (damemi) [inline exporter/collector/metrics_test.go:1442]: another il/is nit

File: .circleci/config.yml (hunk 1 above)
Comment (dashpole) [reply .circleci/config.yml:5]: Generally, most repositories seem to support the latest 2 versions of go, which would be 1.17 and 1.18 for us now.

File: exporter/collector/integrationtest/testcase.go (hunk 2 above)
Comment (dashpole) [reply exporter/collector/integrationtest/testcase.go]: done

File: exporter/collector/internal/datapointstorage/datapointcache.go (hunk 3 above)
Comment (dashpole) [reply exporter/collector/internal/datapointstorage/datapointcache.go]: done

File: exporter/collector/metrics.go (hunk 4 above)
Comment (dashpole) [reply exporter/collector/metrics.go]: done

File: exporter/collector/metrics_test.go (hunk 6 above)
Comment (dashpole) [reply exporter/collector/metrics_test.go]: done

File: exporter/collector/spandata.go [hunk 7]
```diff
@@ -30,12 +30,12 @@ import (
 func pdataResourceSpansToOTSpanData(rs pdata.ResourceSpans) []sdktrace.ReadOnlySpan {
 	resource := rs.Resource()
 	var sds []sdktrace.ReadOnlySpan
-	ilss := rs.InstrumentationLibrarySpans()
+	ilss := rs.ScopeSpans()
```
Comment (damemi) [inline exporter/collector/spandata.go:33]: another nit

**Pull request merged**